src/Blockcore.Indexer.Cirrus/Storage/Mongo/SmartContracts/DaoContractAggregator.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Blockcore.Indexer.Cirrus.Client;
+using Blockcore.Indexer.Cirrus.Storage.Mongo.SmartContracts.Dao;
+using Blockcore.Indexer.Cirrus.Storage.Mongo.Types;
+using Blockcore.Indexer.Core.Client;
+using Blockcore.Indexer.Core.Operations.Types;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+
+namespace Blockcore.Indexer.Cirrus.Storage.Mongo.SmartContracts;
+
+public class DaoContractAggregator : IDAOContractAggregator
+{
+   const string DaoContract = "DAOContract";
+
+   readonly ILogger<DaoContractAggregator> logger;
+   readonly ICirrusMongoDb db;
+   readonly ILogReaderFactory logReaderFactory;
+   readonly CirrusClient cirrusClient;
+
+   public DaoContractAggregator(ILogger<DaoContractAggregator> logger, ICirrusMongoDb mongoData, ILogReaderFactory logReaderFactory, ICryptoClientFactory clientFactory, SyncConnection connection)
+   {
+      db = mongoData;
+      this.logReaderFactory = logReaderFactory;
+      this.logger = logger;
+
+      cirrusClient = clientFactory.Create(connection) as CirrusClient;
+   }
+
+   public async Task<DaoContractComputedTable> ComputeDaoContractForAddressAsync(string address)
+   {
+      var contract = await LookupDaoContractForAddressAsync(address);
+
+      if (contract is null)
+         return null;
+
+      var contractTransactions = await db.CirrusContractTable
+         .AsQueryable()
+         .Where(_ => _.ToAddress == address && _.Success && _.BlockIndex > contract.LastProcessedBlockHeight)
+         .ToListAsync();
+
+      if (contractTransactions.Any())
+      {
+         await AddNewTransactionsDataToDocumentAsync(address, contractTransactions, contract);
+      }
+
+      return contract;
+   }
+
+   async Task<DaoContractComputedTable> LookupDaoContractForAddressAsync(string address)
+   {
+      DaoContractComputedTable contract = await db.DaoContractComputedTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contract is not null)
+         return contract;
+
+      var contractCode = await db.CirrusContractCodeTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contractCode is null || contractCode.CodeType != DaoContract)
+      {
+         logger.LogInformation(
```

Comment:
Do you think we should depend on the contract code?
I guess there is no harm in this however this can be a big payload of data to be fetched from disk maybe unnecessary.

Reply:
If the create transaction exists but is for a standard token etc. we should not try to process it as a DAO contract